src/sentry/static/sentry/app/components/idBadge/userBadge.tsx

```diff
@@ -1,48 +1,73 @@
 import PropTypes from 'prop-types';
 import React from 'react';
 import styled from 'react-emotion';
+import {AvatarUser, Member} from 'app/types';
 import UserAvatar from 'app/components/avatar/userAvatar';
 import Link from 'app/components/links/link';
 import overflowEllipsis from 'app/styles/overflowEllipsis';
 import space from 'app/styles/space';
 import SentryTypes from 'app/sentryTypes';
 import omit from 'lodash/omit';
 
+const defaultProps = {
+  useLink: true,
+  hideEmail: false,
+};
+
+type Props = {
+  avatarSize: UserAvatar['props']['size'];
+  displayName?: string;
+  displayEmail?: string;
+  user?: AvatarUser;
+  member?: Member;
+  orgId?: string;
+} & Partial<typeof defaultProps>;
+
+function getUser(props: {user?: AvatarUser; member?: Member}): AvatarUser | undefined {
+  if (props.user) {
+    return props.user;
+  }
+  if (props.member && props.member.user) {
+    return props.member.user;
+  }
+  return undefined;
+}
+
 const UserBadge = ({
   displayName,
   displayEmail,
-  user,
-  member,
   orgId,
   avatarSize,
   useLink,
   hideEmail,
   ...props
-}) => {
-  const userFromPropsOrMember = user || (member && member.user) || member;
+}: Props) => {
+  const user = getUser(props);
+  const member = props.member;
+  const title =
+    displayName ||
+    (user &&
+      (user.name ||
+        user.email ||
+        user.username ||
+        user.ipAddress ||
+        // Because this can be used to render EventUser models, or User *interface*
+        // objects from serialized Event models. we try both ipAddress and ip_address.
+        user.ip_address)) ||
+    (member && member.name);
+
   return (
-    <StyledUserBadge {...props}>
```

Comment:
Do we have any components that try to style `UserBadge` via `styled`?

Reply:
We do. I'll add the className prop back in.